communication/data_logging.hpp

```diff
@@ -112,8 +104,10 @@ typedef struct
 	
 	uint32_t sys_id;							///< the system ID
 	
+	Console<File>* console;						///< The pointer to the console containing the file to write data to
```

Comment:
In the current state, I did avoid to create a class. 

If you want to create a data_logging class, then you have to (similarly to file_flash):
* create an instance of this class in central_data, and therefore
* call data_logging's constructor in central_data's constructor with a reference to the file, and therefore
* add an instance of file to the board, and therefore
* call file's constructor in the board's constructor with a reference to its name. 

For each file logging you want to add, you have to repeat the process...